libgalois/include/katana/Executor_Deterministic.h

```diff
@@ -91,23 +91,25 @@ template <typename OptionsTy>
 using DItem =
     DItemBase<typename OptionsTy::value_type, OptionsTy::useLocalState>;
 
-class FirstPassBase : public SimpleRuntimeContext {
+class KATANA_EXPORT FirstPassBase : public SimpleRuntimeContext {
 protected:
   bool firstPassFlag;
 
 public:
   explicit FirstPassBase(bool f = true)
       : SimpleRuntimeContext(true), firstPassFlag(f) {}
 
-  bool isFirstPass(void) const { return firstPassFlag; }
+  ~FirstPassBase() override;
 
-  void setFirstPass(void) { firstPassFlag = true; }
+  bool isFirstPass() const { return firstPassFlag; }
 
-  void resetFirstPass(void) { firstPassFlag = false; }
+  void setFirstPass() { firstPassFlag = true; }
+
+  void resetFirstPass() { firstPassFlag = false; }
 
   virtual void alwaysAcquire(Lockable*, katana::MethodFlag) = 0;
 
-  virtual void subAcquire(Lockable* lockable, katana::MethodFlag f) {
+  void subAcquire(Lockable* lockable, katana::MethodFlag f) override {
```

Comment:
Moving from potentially dynamic polymorphism to only static polymorphism seems dangerous. Could you verify that all uses of this class are static? Maybe you could make the superclass private to prevent casts to the super type that could cause problems. (I don't know if this is practical, but it seems worth investigating.)

Reply:
I didn't know what `override` meant. NVM.

Reply:
I chatted with @arthurp offline and we have clarified the issue and it is not relevant here.